src/goto-instrument/code_contracts.cpp

```diff
@@ -282,9 +354,345 @@ const symbolt &code_contractst::new_tmp_symbol(
+    exprt rightPtr = aliasble;
+
+    if(first_iter)
+    {
+      running = same_object(leftPtr, rightPtr);
+      first_iter = false;
+    }
+    else
+    {
+      exprt same = same_object(leftPtr, rightPtr);
+      running = binary_exprt(running, ID_or, same);
+    }
+  }
+
+  return running;
+}
+
+void code_contractst::populate_assigns_references(
+  const symbolt &f_sym,
+  const irep_idt &func_id,
+  goto_programt &created_decls,
+  std::vector<exprt> &created_references)
+{
+  const code_typet &type = to_code_type(f_sym.type);
+  exprt assigns = static_cast<const exprt &>(type.find(ID_C_spec_assigns));
+
+  exprt::operandst &targets = assigns.operands();
+  for(exprt curr_op : targets)
+  {
+    // Create an expression to capture the address of the operand
+    exprt op_addr =
+      exprt(ID_address_of, pointer_type(curr_op.type()), {curr_op});
+
+    // Declare a new symbol to stand in for the reference
+    symbol_exprt standin =
+      new_tmp_symbol(
+        pointer_type(curr_op.type()), f_sym.location, func_id, f_sym.mode)
+        .symbol_expr();
+
+    created_decls.add(goto_programt::make_decl(standin, f_sym.location));
+
+    created_decls.add(goto_programt::make_assignment(
+      code_assignt(standin, std::move(op_addr)), f_sym.location));
+
+    // Add a map entry from the original operand to the new symbol
+    created_references.push_back(standin);
+  }
+}
+
+void code_contractst::instrument_assn_statement(
+  goto_programt::instructionst::iterator &ins_it,
+  goto_programt &program,
+  exprt &assigns,
+  std::vector<exprt> &assigns_references,
+  std::set<exprt> &freely_assignable_exprs)
+{
+  INVARIANT(
+    ins_it->is_assign(),
+    "The first argument of instrument_assn_statement should always be"
+    " an assignment");
+  const exprt &lhs = ins_it->get_assign().lhs();
+  if(freely_assignable_exprs.find(lhs) != freely_assignable_exprs.end())
+  {
+    return;
+  }
+  exprt alias_expr = create_alias_expression(assigns, lhs, assigns_references);
+
+  goto_programt alias_assertion;
+  alias_assertion.add(
+    goto_programt::make_assertion(alias_expr, ins_it->source_location));
+  program.insert_before_swap(ins_it, alias_assertion);
+  std::advance(ins_it, 1);
+}
+
+void code_contractst::instrument_call_statement(
+  goto_programt::instructionst::iterator &ins_it,
+  goto_programt &program,
+  exprt &assigns,
+  std::vector<exprt> &aliasable_references,
+  std::set<exprt> &freely_assignable_exprs)
+{
+  INVARIANT(
+    ins_it->is_function_call(),
+    "The first argument of instrument_call_statement should always be "
+    "a function call");
+  code_function_callt call = ins_it->get_function_call();
+  const irep_idt &called_name =
+    to_symbol_expr(call.function()).get_identifier();
+
+  if(std::strcmp(called_name.c_str(), "malloc") == 0)
+  {
+    aliasable_references.push_back(call.lhs());
+
+    // Make the variable, where result of malloc is stored, freely assignable.
```

Comment:
It is the variable which we create from the call to malloc which is being marked as freely-assignable. Since it is created within the function is should be okay to assign it as long as it is not being de-reference or something. If I remove the first portion of the sentence, it won't be clear what is becoming freely-assignable. Would it be more clear to say "Mark the variable (storing the result of malloc) as freely-assignable"?